js_modules/dagit/packages/core/src/instance/InstanceConfig.tsx

```diff
@@ -1,75 +1,50 @@
+import 'codemirror/addon/search/searchcursor';
+
 import {gql, useQuery} from '@apollo/client';
-import {
-  Box,
-  Colors,
-  HighlightedCodeBlock,
-  Icon,
-  PageHeader,
-  Spinner,
-  Code,
-  Heading,
-  Subheading,
-} from '@dagster-io/ui';
+import {Box, Colors, PageHeader, Spinner, Code, Heading, Subheading} from '@dagster-io/ui';
+import * as codemirror from 'codemirror';
 import * as React from 'react';
-import {Link, useHistory} from 'react-router-dom';
-import styled, {createGlobalStyle, css} from 'styled-components/macro';
+import {createGlobalStyle} from 'styled-components/macro';
 
 import {useQueryRefreshAtInterval, FIFTEEN_SECONDS} from '../app/QueryRefresh';
+import {DagitReadOnlyCodeMirror} from '../ui/DagitCodeMirror';
 
 import {InstanceTabs} from './InstanceTabs';
 import {InstanceConfigQuery} from './types/InstanceConfigQuery';
 
-const YamlShimStyle = createGlobalStyle`
-  .hljs.yaml {
-    margin: 0;
-    padding: 0;
+const InstanceConfigStyle = createGlobalStyle`
+  .react-codemirror2 .CodeMirror.cm-s-instance-config {
+    box-shadow: 0 1px 0 ${Colors.KeylineGray};
+    height: 100%;
   }
 
-  .config-yaml {
-    .hljs-attr {
-      color: ${Colors.Blue700};
-    }
-
-    .hljs-string {
-      color: ${Colors.Green700};
-    }
-
-    .hljs-number {
-      color: ${Colors.Red700};
+  .react-codemirror2 .CodeMirror.cm-s-instance-config {
+    .config-highlight {
+      background-color: ${Colors.Yellow200};
     }
-  }
 `;
 
 export const InstanceConfig = React.memo(() => {
-  const history = useHistory();
   const queryResult = useQuery<InstanceConfigQuery>(INSTANCE_CONFIG_QUERY, {
     fetchPolicy: 'cache-and-network',
   });
-  const [hash, setHash] = React.useState(() => document.location.hash);
+
   const refreshState = useQueryRefreshAtInterval(queryResult, FIFTEEN_SECONDS);
   const {data} = queryResult;
-
-  React.useEffect(() => {
-    // Once data has finished loading and rendering, scroll to hash
-    if (data) {
-      const documentHash = document.location.hash;
-      if (documentHash) {
-        const target = documentHash.slice(1);
-        document.getElementById(target)?.scrollIntoView({
-          block: 'start',
-          inline: 'nearest',
-        });
+  const config = data?.instance.info;
+
+  const onEditorDidMount = (editor: codemirror.Editor) => {
+    const documentHash = document.location.hash;
+    if (documentHash) {
+      const target = new RegExp(`^${documentHash.slice(1)}:`);
+      const cursor = editor.getSearchCursor(target);
+      const found = cursor.findNext();
+      if (found) {
+        editor.markText(cursor.from(), cursor.to(), {className: 'config-highlight'});
+        editor.scrollIntoView(cursor.from());
       }
     }
-  }, [data]);
-
-  React.useEffect(() => {
-    const unlisten = history.listen((location) => {
-      setHash(location.hash);
-    });
-
-    return () => unlisten();
-  }, [history]);
+  };
```

Comment:
So now it will extract the location hash, use CM search to find it, highlight it, and scroll to it. I'm not sure if there's an easy way to put anchors on here, but I think this is good enough for what we need at this point (e.g. link from someplace within Dagit to a known config)